samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -19,36 +19,195 @@
-    // 4. create all input/output/intermediate topics
-    // 5. create the configuration for StreamProcessor
-    // 6. start the StreamProcessor w/ optimized instance of StreamApplication
+  @Override
+  public void run(StreamApplication app) {
+    try {
+      // 1. initialize and plan
+      ExecutionPlan plan = getExecutionPlan(app);
+
+      // 2. create the necessary streams
+      createStreams(plan.getIntermediateStreams());
+
+      // 3. start the StreamProcessors
+      if (plan.getJobConfigs().isEmpty()) {
+        throw new SamzaException("No jobs to run.");
+      }
+      LocalProcessorListener listener = new LocalProcessorListener();
+      plan.getJobConfigs().forEach(jobConfig -> {
+          log.info("Starting job {} StreamProcessor with config {}", jobConfig.getName(), jobConfig);
+          StreamProcessor processor = createStreamProcessor(jobConfig, app, listener);
+          processor.start();
+          processors.add(processor);
+        });
+      appStatus = ApplicationStatus.Running;
+
+      // 4. block until the processors are done or there is a failure
+      awaitComplete();
+
+    } catch (Throwable t) {
+      appStatus = ApplicationStatus.UnsuccessfulFinish;
+      throw new SamzaException("Failed to run application", t);
+    } finally {
+      if (coordination != null) {
+        coordination.reset();
+      }
+    }
   }
 
   @Override
   public void kill(StreamApplication streamApp) {
+    processors.forEach(StreamProcessor::stop);
   }
 
   @Override
   public ApplicationStatus status(StreamApplication streamApp) {
-    return null;
+    return appStatus;
+  }
+
+  /**
+   * Create the Coordination needed by the application runner.
+   * @return {@link CoordinationUtils}
+   * @throws Exception exception
+   */
+  /* package private */ CoordinationUtils getCoordinationUtils() throws Exception {
+    ApplicationConfig appConfig = new ApplicationConfig(config);
+    String clazz = appConfig.getCoordinationServiceFactoryClass();
+    if (clazz != null) {
+      CoordinationServiceFactory factory = ClassLoaderHelper.fromClassName(clazz);
+      String groupId = String.format("app-%s-%s", appConfig.getAppName(), appConfig.getAppId());
+      return factory.getCoordinationService(groupId, uid, config);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Create intermediate streams.
+   * @param intStreams intermediate {@link StreamSpec}s
+   * @throws Exception exception
+   */
+  /* package private */ void createStreams(List<StreamSpec> intStreams) throws Exception {
+    if (!intStreams.isEmpty()) {
+      if (coordination != null) {
+        Latch initLatch = coordination.getLatch(1, LATCH_INIT);
+        coordination.getLeaderElector().tryBecomeLeader(() -> {
+            getStreamManager().createStreams(intStreams);
+            initLatch.countDown();
+          });
+        initLatch.await(LATCH_TIMEOUT_MINUTES, TimeUnit.MINUTES);
```

Comment:
Seems like all non-leader LocalRunners are going to wait for 10 minutes on startup here. Am I misunderstanding something?